api/authenticator.py

```diff
@@ -1194,16 +1194,22 @@ class BasicAuthenticationProvider(AuthenticationProvider):
     SETTINGS = [
         { "key": TEST_IDENTIFIER,
           "label": _("Test Identifier"),
-          "description": _("A valid identifier that can be used to test that patron authentication is working.") },
-        { "key": TEST_PASSWORD, "label": _("Test Password"), "description": _("The password for the test identifier.") },
+          "description": _("A valid identifier that can be used to test that patron authentication is working."),
+          "optional": True,
```

Comment:
This is already the BasicAuthenticationProvider, so you don't need to make it optional for OAuth providers...